### PR TITLE
RFR: Set action status as failure on python action exceptions

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
* Print the exception to stderr and make sure exception shows in
action execution object.

### CODE REVIEW

File: st2actions/st2actions/runners/python_action_wrapper.py [hunk 1]
```diff
@@ -54,17 +54,28 @@ def __init__(self, pack, file_path, parameters=None):
...
-        output = action.run(**self._parameters)
+
+        stream = None
+        try:
+            # XXX: We need a way for actions to return status and result.
+            output = action.run(**self._parameters)
+            stream = sys.stdout
+            exit_code = 0
+        except Exception as e:
+            output = e
```
Comment (manasdk) [inline st2actions/st2actions/runners/python_action_wrapper.py]: output = traceback.format_exc()

File: st2actions/st2actions/runners/pythonrunner.py [hunk 2]
```diff
@@ -148,11 +148,13 @@ def run(self, action_parameters):
         stdout, stderr = process.communicate()
         exit_code = process.returncode
 
-        if ACTION_OUTPUT_RESULT_DELIMITER in stdout:
-            split = stdout.split(ACTION_OUTPUT_RESULT_DELIMITER)
+        stream_to_process = stderr if exit_code else stdout
```
Comment (manasdk) [inline st2actions/st2actions/runners/pythonrunner.py]: Both streams should be processes in like localshellrunner.

**Comment by manasdk** [comment]
-1

File: st2actions/st2actions/runners/pythonrunner.py (hunk 2 above)
Comment (lakshmi-kannan) [inline st2actions/st2actions/runners/pythonrunner.py]: Discussed offline. For now, either the action returns a result or an exception. So this is not needed.

**Comment by manasdk** [comment]
+1

File: st2actions/st2actions/runners/pythonrunner.py [hunk 3]
```diff
@@ -148,11 +148,13 @@ def run(self, action_parameters):
...
+
+        if ACTION_OUTPUT_RESULT_DELIMITER in stream_to_process:
+            split = stream_to_process.split(ACTION_OUTPUT_RESULT_DELIMITER)
             assert len(split) == 3
             result = split[1].strip()
-            stdout = split[0] = split[2]
+            stream_to_process = split[0] = split[2]
```
Comment (Kami) [inline st2actions/st2actions/runners/pythonrunner.py]: Slightly confused what's going on here - beyond this point `stream_to_process` is unused?

**Pull request merged**